spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -734,6 +734,22 @@ class CometSparkSessionExtensions
       } else {
         var newPlan = transform(plan)
 
+        // if the plan cannot be run natively then explain why (when appropriate config is enabled)
+        if (CometConf.COMET_EXPLAIN_FALLBACK_ENABLED.get() && !isCometNative(newPlan)) {
```

Comment:
Comet could only trigger native execution for partial query plan. This check `isCometNative(newPlan)` only applies for the top operator.